cou/apps/auxiliary.py

```diff
@@ -53,31 +53,31 @@ def is_valid_track(self, charm_channel: str) -> bool:
             logger.debug("'%s' has been installed from the charm store", self.name)
             return True
 
-        track = self._get_track_from_channel(charm_channel)
-        return (self.charm, self.series, track) in TRACK_TO_OPENSTACK_MAPPING
+        current_track = self._get_track_from_channel(charm_channel)
+        possible_tracks = OPENSTACK_TO_TRACK_MAPPING.get(
+            (self.charm, self.series, self.current_os_release.codename), []
+        )
+        return (
+            self.charm,
+            self.series,
+            current_track,
+        ) in TRACK_TO_OPENSTACK_MAPPING and len(possible_tracks) > 0
 
     @property
-    def possible_current_channels(self) -> list[str]:
-        """Return the possible current channels based on the series and current OpenStack release.
+    def expected_current_channel(self) -> str:
+        """Return the expected current channel.
 
-        :return: The possible current channels for the application.
-        :rtype: list[str]
-        :raises ApplicationError: When cannot find tracks.
+        Expected current channel is the channel that the application is suppose to be using based
+        on the current series, workload version and, by consequence, the OpenStack release
+        identified.
+        :return: The expected current channel of the application. E.g: "3.9/stable"
+        :rtype: str
         """
-        tracks = OPENSTACK_TO_TRACK_MAPPING.get(
+        *_, track = OPENSTACK_TO_TRACK_MAPPING[
```

Comment:
not-blocking: I know that we are using `is_valid_track` before these, but if I call in another place without checking if track is valid, it can raise ValueError, and I think we should mention it in docstring.